### ISSUE TITLE
Add deprecation warning to validation command

### ISSUE THREAD

**Issue by chell0veck** [hashicorp/terraform#33087]
### Terraform Version

```shell
1.1.2
```

### Use Cases

During upgrade module from `0.14.11` to `1.1.2` have catch the error 
`
│ Call to function "map" failed: the "map" function was deprecated in
│ Terraform v0.12 and is no longer available; use tomap({ ... }) syntax to
│ write a literal map.
`
What happened is that this evaluation is context aware and having complicated conditional logic will only occur when functionality triggered.

Basically I've picked up a wrong (not full) set of inputs for testing.

### Attempted Solutions

Actually no - I didn't receive `deprecation` warning cause choose as wrong set in inputs during migration planning

### Proposal

Including unaware `deprecation` warning  into `validate` command would help to prevent such `context related` things and definitely better plan the upgrade

### References

can not share that
